enumerable.rb

```diff
@@ -0,0 +1,119 @@
+# Enumerable method
+module Enumerable
+  def my_each
+    arr = self
+    if arr.is_a?(Array)
+      (0..(arr.length - 1)).each do |i|
+        yield(arr[i])
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index
+    arr = self
+    (0..(arr.length - 1)).each do |i|
+      yield(arr[i], i)
+    end
+    arr
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    puts result.to_s
+    result
+  end
+
+  def my_all?
+    my_each do |num|
+      if yield(num) == false
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
+
+  def my_any?
+    my_each do |num|
+      if yield(num) == true
+        puts true
+        return true
+      end
+    end
+    puts false
+    false
+  end
+
+  def my_none?
+    my_each do |num|
+      if yield(num) == true
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
+
+  def my_count
+    count = 0
+    my_each do |_acc|
+      count += 1
+    end
+    puts count
+    count
+  end
```

Comment:
- #my_count returns the number of items in enum through enumeration in a range
- #my_count counts the number of items in enum that are equal to item if an argument is given
- #my_count counts the number of elements yielding a true value if a block is given